### PR TITLE
c-lib: Support to build a static archive

### PR DESCRIPTION
**Pull request by udzura**
This will be useful to bind libcriu features into one binary. e.g. container runtime or some ops tools.

Would anyone have time to check this out?

### CODE REVIEW

File: lib/Makefile [hunk 1]
```diff
@@ -19,8 +20,12 @@ ldflags-so		+= -lprotobuf-c
 lib/c/$(CRIU_SO): lib/c/built-in.o
 	$(call msg-link, $@)
 	$(Q) $(CC) -shared $(cflags-so) -o $@ $^ $(ldflags-so) $(LDFLAGS)
+lib/c/$(CRIU_A): lib/c/built-in.o
+	$(call msg-link, $@)
+	$(Q) $(AR) rcs $@ $^
 lib-c: lib/c/$(CRIU_SO)
-.PHONY: lib-c
+lib-a: lib/c/$(CRIU_A)
```
Comment (0x7f454c46) [inline lib/Makefile]: don't you need to add it to `all-y` to build .a?

Comment (cyrillos) [reply lib/Makefile]: yes, it should be added into `all-y` target in the same file

Comment (0x7f454c46) [reply lib/Makefile]: Probably, it's also worth to add `install` rule if we plan providing .a library.

Comment (udzura) [reply lib/Makefile]: Thabjs for feedback! I've added a commit.

**Comment by avagin** [comment]
ping @0x7f454c46 @cyrillos

**Comment by cyrillos** [comment]
ack

**Comment by avagin** [comment]
@udzura pls take a look what fails in travis:
https://travis-ci.org/checkpoint-restore/criu/jobs/503457651

**Comment by rst0git** [comment]
> @udzura pls take a look what fails in travis:

The failure is because `lib/c/libcriu.a` was not removed after `make clean`. This should fix it.

```diff
diff --git a/lib/Makefile b/lib/Makefile
index 7b95956a..4a131e88 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -42,7 +42,7 @@ clean-lib:
        $(Q) $(MAKE) $(build)=lib/py clean
 .PHONY: clean-lib
 clean: clean-lib
-cleanup-y      += lib/c/$(CRIU_SO) lib/c/criu.pc
+cleanup-y      += lib/c/$(CRIU_SO) lib/c/$(CRIU_A) lib/c/criu.pc
 mrproper: clean

 install: lib-c lib-py crit/crit lib/c/criu.pc.in
```

**Comment by udzura** [comment]
Thanks for advice!! Sorry for late to reply. I will try to rebase.

**Comment by rst0git** [comment]
@udzura thank you for updating the PR!

Would it be possible also to add `CRIU_A` to the `install` and `uninstall` phony targets?

**Comment by udzura** [comment]
Pushed dbe4527 . I will revert this commit if it is judged to be unnecessary.

**Comment by cyrillos** [comment]
Looks ok to me

**Comment by avagin** [comment]
Applied. Thanks!

**Pull request merged**